demo/admin/src/App.tsx

```diff
@@ -100,58 +101,68 @@ export function App() {
-                                                                                    <SitePreview
-                                                                                        resolvePath={(path: string, scope) => {
-                                                                                            return `/${scope.language}${path}`;
-                                                                                        }}
-                                                                                        {...props}
-                                                                                    />
-                                                                                )}
-                                                                            />
-                                                                            <Route
-                                                                                render={() => (
-                                                                                    <MasterLayout
-                                                                                        headerComponent={MasterHeader}
-                                                                                        menuComponent={AppMasterMenu}
-                                                                                    >
-                                                                                        <MasterMenuRoutes menu={masterMenuData} />
-                                                                                    </MasterLayout>
-                                                                                )}
-                                                                            />
-                                                                        </Switch>
-                                                                    )}
-                                                                </ContentScopeProvider>
-                                                            </RouterBrowserRouter>
-                                                        </CurrentUserProvider>
-                                                    </CmsBlockContextProvider>
-                                                </SnackbarProvider>
-                                            </DndProvider>
-                                        </MuiThemeProvider>
+                                        <BlocksConfigProvider
+                                            supportsBlock={(name, scope) => {
+                                                if (scope.domain === "main") {
+                                                    return true;
+                                                } else {
+                                                    return name !== "NewsDetail" && name !== "NewsList" && name !== "NewsLink";
```

Comment:
Suggestion from @jamesricky: Don't hardcode the block names here, use Block.name instead.